eugl/fmask.py

```diff
@@ -197,13 +202,55 @@ def _sentinel2_fmask(dataset_path, container, granule, out_fname, work_dir):
 
     # run fmask
     cmd = ["fmask_sentinel2Stacked.py", "-a", vrt_fname, "-z", angles_fname,
-           "-o", out_fname]
+           "-o", out_fname,
+           "--cloudbufferdistance", str(cloud_buffer_distance),
+           "--shadowbufferdistance", str(cloud_shadow_buffer_distance)]
+
+    if parallax_test:
+        cmd.extend("--parallaxtest")
+
     run_command(cmd, work_dir)
 
 
-def fmask(dataset_path, granule, out_fname, outdir, acq_parser_hint=None):
+def fmask(dataset_path, granule, out_fname, outdir, acq_parser_hint=None,
+          cloud_buffer_distance=150.0, cloud_shadow_buffer_distance=300.0,
+          parallax_test=False):
     """
     Execute the fmask process.
+
+    :param dataset_path:
+    :type str:
+        A str containing the full file pathname to the dataset.
```

Comment:
Cool thanks for that.
I'll go for the first form.  Compact form looks a little odd.

Reply:
on the ```dataset_path, outdir, out_fname```, dataset_path could be a file or a directory. out_fname is supposed to be a filename, and outdir ... i'll update that.

Reply:
> out_fname is supposed to be a filename

Ah, it looks like you're right. I was judging it by the docstring, which says otherwise:  "out_fname: A str containing the full file pathname to contain the output file."

Reply:
English is my first language ... dang no excuse.  I'll make the docstring clearer

Reply:
I've updated the docstring to be explicit by stating *fully qualified name* to get around the ambiguity of filename and pathname.